### DISCUSSION TITLE
What are your priorities in terms of documentation?

### DISCUSSION THREAD

**Discussion by mdouze** [facebookresearch/faiss#1616] [General]
Dear Faiss users,
In 2021 we are planning to work on improvements of the Faiss documentation. Please comment here to say what you think is most important for your usage. Different axes we are thinking of:

1. better tutorial
2. better information on how to compile or install Faiss (please be specific)
3. Python docstrings for interactive use in notebooks
4. doxygen documentation: make more accessible, improve the doxygen annotations
5. definition of what is the public / private API, where public is guaranteed not to change
6. more structured presentation of the index types
7. more benchmarks
8. detail existing topics in the wiki or add other topics (eg. Faiss as a database)
9. explicit roadmap
10. help for porting to other platforms
11. C API documentation
12. an interactive forum to get help from (like https://discuss.pytorch.org/)
13. anything else?

We don't make any promise, but we will try to take feedback into account!

**Comment by OMGTT** [comment]
6.more structured presentation of the index types

**Comment by aguerreb** [comment]
6.more structured presentation of the index types

**Comment by SamsTheGreatest** [comment]
3. Python docstrings for interactive use in notebooks.
Or is there any documentation available already? I find it hard to explore/find needed functionality
